deps/config/manifests/vault.yml.template

```diff
@@ -0,0 +1,49 @@
+- kind: vault
+  type: hashicorp
+  name: hashicorp-kv-v2
+  specs:
+    # allowed_tenants: [tenant1, tenant2]
```

Comment:
I thought we agreed on moving allowed tenants to first level since it wont be part of the user request payload. Also for simplification as you would have only specs, instead of specs+ config